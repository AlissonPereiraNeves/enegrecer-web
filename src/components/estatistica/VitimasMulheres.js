import React, { Component } from 'react';
import firebaseApp from '../../utils/firebaseUtils';
import TipoDeEstatistica from './TipoDeEstatistica';

export default class VitimasMulheres extends Component {
  constructor(props) {
    super(props);
    this.state = { percentagem: '0' };
  }

  componentDidMount() {
    const ref = firebaseApp.database().ref();
    let totalVitimas = 0;
    let totalMulheresVitmas = 0;
    const query = ref.child('pessoasEnvolvidas').orderByKey();
    query.once('value')
      .then((snapshot) => {
        snapshot.forEach((denunciaSnapshot) => {
          const vitima = denunciaSnapshot.val().vitima;
          if (vitima) {
            totalVitimas += 1;
            if (vitima.genero === 'Feminino') {
              totalMulheresVitmas += 1;
            }
          }
        });
        let percentagem = (totalMulheresVitmas * 100) / totalVitimas;
        percentagem = parseFloat(percentagem.toFixed(2));
        this.setState({ percentagem: `${percentagem}%` });
      });
  }

  render() {
    return (<TipoDeEstatistica
      id="verde"
      numero={this.state.percentagem}
      categoria="Vitimas Mulheres"
    />
    );
  }
}
