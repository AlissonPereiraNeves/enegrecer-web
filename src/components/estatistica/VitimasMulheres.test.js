import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import VitimasMulheres from './VitimasMulheres';

describe('Vitimas Mulheres', () => {
  it('deve construir a pagina corretamente ', () => {
    const vitimasMulheres = shallow(<VitimasMulheres />);
    const json = toJson(vitimasMulheres);
    expect(json).toMatchSnapshot();
  });
});
