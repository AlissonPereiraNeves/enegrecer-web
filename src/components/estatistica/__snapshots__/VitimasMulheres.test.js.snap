// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vitimas Mulheres deve construir a pagina corretamente  1`] = `
<TipoDeEstatistica
  categoria="Vitimas Mulheres"
  id="verde"
  numero="0"
/>
`;
